### PR TITLE
Topic/dev integration tests

### PR DESCRIPTION
**Pull request by Asakiz**

### CODE REVIEW

File: updatehub/tests/common.rs [hunk 1]
```diff
@@ -10,6 +10,12 @@ use std::{env, path::PathBuf};
 pub enum FakeServer {
     NoUpdate,
     HasUpdate(String),
+    CheckRequirementsTest(String),
+}
+
+pub enum StopMessage {
+    Custom(String),
+    Polling(Polling),
```
Comment (Jonathas-Conceicao) [inline updatehub/tests/common.rs:18]: This changes to `FakeServer` and `StopMessage`, and related changes here, doesn't seem to be necessary in this particular commit. If possible it should be moved to the next one where a test with failing checking requirements is actually added

File: updatehub/tests/common.rs [hunk 2]
```diff
@@ -169,7 +164,7 @@ pub fn create_mock_server(server: FakeServer) -> Vec<Mock> {
                     "target": "/dev/device1",
                     "sha256sum": "bdbdb660c6b7b18a8376695ff18842d1ea9f3c2c80a192596226b5191bea42bf",
                     "size": 120,
-                    "force_check_requirements_fail" : false
+                    "force-check-requirements-fail": false
```
Comment (Jonathas-Conceicao) [inline updatehub/tests/common.rs:167]: Fixes like this one should't be here

File: updatehub/tests/successful_integration_test.rs [hunk 3]
```diff
@@ -11,7 +11,7 @@ use common::{
 pub mod common;
 
 #[test]
-fn correct_config_no_update_no_polling() {
```
Comment (Jonathas-Conceicao) [inline updatehub/tests/successful_integration_test.rs:14]: The redundancy here is actually useful. It helps us run tests by partially matching the test names. This commit should be reworked in order to rename all integration tests to mach the following use cases:

- Match all correct usage logging tests;
- Match all failing usage logging tests;
- Match all with api calls;
- Match all tests without api calls;

Either test names or submodules can be use to make this matching possibles.

File: updatehub/tests/successful_integration_test.rs [hunk 4]
```diff
@@ -313,9 +321,8 @@ fn correct_config_update_no_polling_probe_api() {
     <timestamp> DEBG 70% of the file has been downloaded
     <timestamp> DEBG 90% of the file has been downloaded
     <timestamp> DEBG 100% of the file has been downloaded
-    <timestamp> INFO installing update: 4304291fa4d86ba6f924b3385ee1c0d5b4a0f8985abf3b84df73da3e5182ff0b
+    <timestamp> INFO installing update: 14aa3cb34ccd7047b81316ba42e5766b8eaeb2ddb32d537856e2819aacdf8a06
     <timestamp> INFO using installation set as target 1
-    <timestamp> DEBG running default check_requirements
```
Comment (otavio) [inline updatehub/tests/successful_integration_test.rs:318]: Is this intended? this does not seem related to this change.

Comment (Asakiz) [reply updatehub/tests/successful_integration_test.rs:318]: Yes, as intended. Because now the object has his own implementation of check requirements.

File: updatehub/tests/common.rs [hunk 5]
```diff
@@ -202,20 +204,20 @@ pub fn create_mock_server(server: FakeServer) -> Vec<Mock> {
...
                 .with_body(std::iter::repeat(0xF).take(120).collect::<Vec<_>>())
             .create(),
-        ],
+        ]
     }
 }
 
 pub fn format_output_server(s: String) -> (String, String) {
-    let s = remove_carriage_newline_caracters(remove_timestamp(remove_version(s)));
+    let s = remove_carriage_newline_characters(remove_timestamp(remove_version(s)));
```
Comment (otavio) [inline updatehub/tests/common.rs:279]: This should be done in another commit as it a typo fix.

File: updatehub/tests/common.rs [hunk 6]
```diff
@@ -202,20 +204,20 @@ pub fn create_mock_server(server: FakeServer) -> Vec<Mock> {
...
+                format!("/products/{}/packages/14aa3cb34ccd7047b81316ba42e5766b8eaeb2ddb32d537856e2819aacdf8a06/objects/bdbdb660c6b7b18a8376695ff18842d1ea9f3c2c80a192596226b5191bea42bf", product_uid)
                     .as_str(),
             )
             .match_header("Content-Type", "application/json")
             .match_header("Api-Content-Type", "application/vnd.updatehub-v1+json")
             .with_status(200)
                 .with_body(std::iter::repeat(0xF).take(120).collect::<Vec<_>>())
             .create(),
-        ],
+        ]
```
Comment (otavio) [inline updatehub/tests/common.rs]: This is not related.

File: updatehub/tests/common.rs [hunk 7]
```diff
@@ -253,7 +255,7 @@ pub fn remove_timestamp(s: String) -> String {
     date_re.replace_all(&s, "<timestamp> ").to_string()
 }
 
-pub fn remove_carriage_newline_caracters(s: String) -> String {
+pub fn remove_carriage_newline_characters(s: String) -> String {
```
Comment (otavio) [inline updatehub/tests/common.rs:317]: This should be done in another commit as it a typo fix.

File: updatehub/tests/common.rs [hunk 8]
```diff
@@ -266,5 +268,5 @@ pub fn format_output_client_log(s: String) -> String {
     let tmpfile_re = Regex::new(r#"\\"/tmp/.tmp.*""#).expect("fail to compile the tmpfile regexp");
     let s = tmpfile_re.replace_all(&s, r#""<file>""#);
 
-    remove_carriage_newline_caracters(s.to_string())
+    remove_carriage_newline_characters(s.to_string())
```
Comment (otavio) [inline updatehub/tests/common.rs:330]: This should be done in another commit as it a typo fix.

File: updatehub/tests/successful_integration_test.rs [hunk 9]
```diff
@@ -4,7 +4,7 @@
 
 use common::{
     create_mock_server, format_output_client_log, format_output_server, get_output_server,
-    remove_carriage_newline_caracters, run_client_log, run_client_probe, FakeServer, Polling,
+    remove_carriage_newline_characters, run_client_log, run_client_probe, FakeServer, Polling,
```
Comment (otavio) [inline updatehub/tests/successful_integration_test.rs:7]: This should be done in another commit as it a typo fix.

File: updatehub/tests/successful_integration_test.rs [hunk 10]
```diff
@@ -164,7 +164,7 @@ fn correct_config_no_update_polling_probe_api() {
     <timestamp> DEBG moving to Probe state after delay.
     "###);
 
-    insta::assert_snapshot!(remove_carriage_newline_caracters(output_client), @r###"
+    insta::assert_snapshot!(remove_carriage_newline_characters(output_client), @r###"
```
Comment (otavio) [inline updatehub/tests/successful_integration_test.rs:167]: This should be done in another commit as it a typo fix.

File: updatehub/tests/successful_integration_test.rs [hunk 11]
```diff
@@ -233,7 +233,7 @@ fn correct_config_no_update_no_polling_probe_api() {
     <timestamp> DEBG staying on Park state.
     "###);
 
-    insta::assert_snapshot!(remove_carriage_newline_caracters(output_client), @r###"
+    insta::assert_snapshot!(remove_carriage_newline_characters(output_client), @r###"
```
Comment (otavio) [inline updatehub/tests/successful_integration_test.rs:236]: This should be done in another commit as it a typo fix.

File: updatehub/tests/successful_integration_test.rs (hunk 4 above)
Comment (otavio) [reply updatehub/tests/successful_integration_test.rs:318]: Ahhh, makes sense.

File: updatehub/tests/successful_integration_test.rs [hunk 12]
```diff
@@ -325,15 +332,7 @@ fn correct_config_update_no_polling_probe_api() {
...
-    insta::assert_snapshot!(output_server_info_2.trim(), @r###"
-    <timestamp> INFO installing update: 4304291fa4d86ba6f924b3385ee1c0d5b4a0f8985abf3b84df73da3e5182ff0b
-    <timestamp> INFO using installation set as target 1
-    <timestamp> INFO swapping active installation set
-    <timestamp> INFO update installed successfully
-    <timestamp> INFO triggering reboot
-    "###);
-
-    insta::assert_snapshot!(remove_carriage_newline_caracters(output_client), @r###"
+    insta::assert_snapshot!(remove_carriage_newline_characters(output_client), @r###"
```
Comment (otavio) [inline updatehub/tests/successful_integration_test.rs:335]: This should be done in another commit as it a typo fix.

File: updatehub/tests/successful_integration_test.rs [hunk 13]
```diff
@@ -302,6 +302,14 @@ fn correct_config_update_no_polling_probe_api() {
     <timestamp> DEBG staying on Park state.
     "###);
 
+    insta::assert_snapshot!(output_server_info_2.trim(), @r###"
+    <timestamp> INFO installing update: 14aa3cb34ccd7047b81316ba42e5766b8eaeb2ddb32d537856e2819aacdf8a06
+    <timestamp> INFO using installation set as target 1
+    <timestamp> INFO swapping active installation set
+    <timestamp> INFO update installed successfully
+    <timestamp> INFO triggering reboot
+    "###);
```
Comment (otavio) [inline updatehub/tests/successful_integration_test.rs:311]: This seems to be changing the code place. Why?

Comment (Asakiz) [reply updatehub/tests/successful_integration_test.rs:311]: I changed this intentionally on next commit, because all the other tests follow the same order: info log then probe log.

Comment (otavio) [reply updatehub/tests/successful_integration_test.rs:311]: Good, but then move this to a specific commit as it allows you to add a proper commit log explaining the reason for the move.

File: updatehub/tests/failed_integration_test.rs [hunk 14]
```diff
@@ -275,14 +275,7 @@ fn failing_fail_check_requirements() {
     <timestamp> DEBG staying on Park state.
     "###);
 
-    insta::assert_snapshot!(output_server_info_2.trim(), @r###"
-    <timestamp> INFO installing update: fb21b217cb83e8af368c773eb13bad0a94e1b0088c6bf561072decf3c1ae9df3
-    <timestamp> INFO using installation set as target 1
-    <timestamp> ERRO error state reached: fail to check the requirements
-    <timestamp> INFO returning to machine's entry point
-    "###);
-
```
Comment (Jonathas-Conceicao) [inline updatehub/tests/failed_integration_test.rs]: I belive this removed lines should be on the previous commit
